engine/src/main/java/org/terasology/rendering/dag/RenderTaskListGenerator.java

```diff
@@ -227,4 +228,25 @@ public void refresh() {
         generateFrom(nodeList);
     }
 
+    /**
+     * Refreshes the task list if a refresh has been requested.
+     */
+    public void refreshIfNeeded() {
+        if (refreshRequested) {
+            refresh();
+
+            // Clear refresh requests to prevent refreshing unless needed
+            refreshRequested = false;
+        }
+    }
```

Comment:
These changes break the `ConditionDependentNode` class, which now requires a `WorldRenderer` reference.